thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/datasource/sql/SqlThirdEyeDataSource.java

```diff
@@ -0,0 +1,204 @@
+  }
+
+  @Override
+  public ThirdEyeResponse execute(ThirdEyeRequest request) throws Exception {
+    LinkedHashMap<MetricFunction, List<ThirdEyeResultSet>> metricFunctionToResultSetList = new LinkedHashMap<>();
+
+    TimeSpec timeSpec = null;
+    try {
+      for (MetricFunction metricFunction : request.getMetricFunctions()) {
+        String dataset = metricFunction.getDataset();
+        DatasetConfigDTO datasetConfig = ThirdEyeUtils.getDatasetConfigFromName(dataset);
+        TimeSpec dataTimeSpec = ThirdEyeUtils.getTimestampTimeSpecFromDatasetConfig(datasetConfig);
+
+        if (timeSpec == null) {
+          timeSpec = dataTimeSpec;
+        }
+
+        String[] tableComponents = dataset.split("\\.");
+        String sourceName = tableComponents[0];
+        String dbName = tableComponents[1];
+
+        String sqlQuery = SqlUtils.getSql(request, metricFunction, request.getFilterSet(), dataTimeSpec, sourceName);
+        ThirdEyeResultSet thirdEyeResultSet = executeSQL(new SqlQuery(sqlQuery, sourceName, dbName,
+            metricFunction.getMetricName(), request.getGroupBy(), request.getGroupByTimeGranularity(), dataTimeSpec));
+
+        List<ThirdEyeResultSet> resultSetList = new ArrayList<>();
+        resultSetList.add(thirdEyeResultSet);
+
+        metricFunctionToResultSetList.put(metricFunction, resultSetList);
+
+      }
+      List<String[]> resultRows = ThirdEyeResultSetUtils.parseResultSets(request, metricFunctionToResultSetList, "SQL");
+
+      return new RelationalThirdEyeResponse(request, resultRows, timeSpec);
+    } catch (Exception e) {
+      throw e;
+    }
+
+  }
+
+  /**
+   * Returns the cached ResultSetGroup corresponding to the given Presto query.
+   *
+   * @param SQLQuery the query that is specifically constructed for Presto.
+   * @return the corresponding ThirdEyeResultSet to the given Presto query.
+   *
+   */
+  private ThirdEyeResultSet executeSQL(SqlQuery SQLQuery) throws Exception {
+    ThirdEyeResultSet thirdEyeResultSet;
+    try {
+      thirdEyeResultSet = sqlResponseCache.get(SQLQuery);
+    } catch (Exception e) {
+      throw e;
+    }
+    return thirdEyeResultSet;
+  }
+
+  @Override
+  public List<String> getDatasets() throws Exception {
+    return CACHE_REGISTRY_INSTANCE.getDatasetsCache().getDatasets();
+  }
+
+  @Override
+  public void clear() throws Exception {
+    // left blank
+  }
+
+  @Override
+  public void close() throws Exception {
+    // left blank
+  }
+
+  @Override
+  public long getMaxDataTime(String dataset) throws Exception {
+    LOG.info("Getting max data time for " + dataset);
+    return sqlResponseCacheLoader.getMaxDataTime(dataset);
+  }
+
+  @Override
+  public Map<String, List<String>> getDimensionFilters(String dataset) throws Exception {
+    LOG.info("Running dimension filters for " + dataset);
+    return this.sqlResponseCacheLoader.getDimensionFilters(dataset);
+  }
+
+
+  private static LoadingCache<SqlQuery, ThirdEyeResultSet> buildResponseCache(
```

Comment:
Can we reuse these methods from the PinotThirdEyeDataSource? We can discuss offline on how to design this.

Reply:
Thanks for the suggestion. So I have refactored and moved buildResponseCache function into ThirdEyeUtils. (I had to create a superclass called RelationalQuery for SqlQuery and PinotQuery to extend to reuse this function.